File: classifier/data/raw/ai/samples/project_euler_java/Problem036.java
/** Project Euler 36: sum numbers below one million palindromic in base 10 and base 2. */
public class Problem036 {
    public static void main(String[] args) {
        int sum = 0;
        for (int n = 1; n < 1_000_000; n++) {
            if (isPalindrome(Integer.toString(n))
                    && isPalindrome(Integer.toBinaryString(n))) {
                sum += n;
            }
        }
        System.out.println(sum);
    }

    private static boolean isPalindrome(String value) {
        for (int left = 0, right = value.length() - 1; left < right; left++, right--) {
            if (value.charAt(left) != value.charAt(right)) {
                return false;
            }
        }
        return true;
    }
}

File: classifier/data/raw/human/samples/project_euler_zig/problem_036.zig
const std = @import("std");

// Project Euler 36: sum numbers palindromic in both base ten and base two.
pub fn main() !void {
    var sum: u32 = 0;
    var n: u32 = 1;
    while (n < 1_000_000) : (n += 1) {
        if (isPalindrome(n, 10) and isPalindrome(n, 2)) sum += n;
    }
    try printAnswer(sum);
}

fn isPalindrome(number: u32, base: u32) bool {
    var n = number;
    var reversed: u32 = 0;
    while (n > 0) : (n /= base) reversed = reversed * base + n % base;
    return number == reversed;
}

fn printAnswer(value: anytype) !void {
    var buffer: [64]u8 = undefined;
    const text = try std.fmt.bufPrint(&buffer, "{d}\n", .{value});
    try std.Io.File.stdout().writeStreamingAll(std.Options.debug_io, text);
}
